db/app.js

```diff
@@ -1,15 +1,31 @@
-const express = require("express")
-const app = express()
-const { getTopics, getEndpoints, getArticleById } = require("./controllers")
+const express = require("express");
+const app = express();
+const { getTopics, getEndpoints, getArticleById } = require("./controllers");
 
-app.use(express.json())
+app.use(express.json());
 
-app.get(`/api/topics`, getTopics)
-app.get(`/api`, getEndpoints)
-app.get(`/api/articles/:article_id`, getArticleById)
+app.get(`/api/topics`, getTopics);
+app.get(`/api`, getEndpoints);
+app.get(`/api/articles/:article_id`, getArticleById);
+
+app.use((err, req, res, next) => {
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
+});
+
+app.use((err, req, res, next) => {
+    if (err.code === "22P02") {
+      res.status(400).send({ msg: "Invalid Request" });
+    } else {
+      next(err);
+    }
+  });
 
 app.all("*", (req, res) => {
-    res.status(404).send("404 - Not Found")
-})
+  res.status(404).send("404 - Not Found");
+});
 
```

Comment:
We also want to add something to handle error 500 (server error) this would be the final error (you can't test for this)

Reply:
would that be something like another app.use if status(500) send a message about it, else next(err)?